R/api.R

```diff
@@ -16,108 +16,360 @@
   get_content(hn_api_response)
 }
 
-#' Retrieves the item corresponding to the specified id
+
+#' Checks whether max_items is correctly defined. If not throws an error
 #'
-#' @export
+#' @param max_items maximum number of ids to retrieve
+#
+.assert_max_items <- function(max_items) {
+  if (!is.null(max_items)) {
+    assert(is.numeric(max_items), "max_items must be a numeric type")
+    assert(max_items > 0,
+           "max_items must be greater than 0")
+  }
+}
+
+#' Checks whether ids are correctly defined. If not throws an error
+#'
+#' @param ids maximum number of ids to retrieve
+#
+.assert_ids <- function(ids) {
+    assert(all(is.numeric(ids)), "every id must be a numeric type")
+    assert(all(ids > 0),
+           "every id must be greater than 0")
+}
+
```

Comment:
`api.R` is starting to get a bit big. Maybe we should move those to a separate file something like `assertion.R`. What do you think?

Reply:
dots have to be removed, right? how about `assert` function from utils, I think we should also move it

Reply:
You are right!